powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/ComputeOfflineSignatureStep.java

```diff
@@ -167,8 +170,9 @@ private String calculateOfflineSignature(final String offlineData, final StepLog
         }
         final String operationId = parts[0];
         final String operationData = parts[3];
-        final String nonce = parts[5];
+        final String nonce = parts[parts.length - 2];
         final String signatureLine = parts[parts.length - 1];
+        final String otp = parts.length == 8 ? parts[parts.length - 3] : null;
```

Comment:
🤔 Is there any way to make it more robust in the future, if we add another attribute after OTP?

Reply:
I was thinking about joining all the unexpected attributes together (indices 5 to len-3). However if there is a new attribute in offline qr code it does not have to be part of the signature base necessarily (such as a new informative string for a user).

But you are right about robustness. I came with a new parsing rule in the new commit. However, there is still a hypothetical scenario when the solution is not robust enough. (Adding a new numerical attribute and allowing not to use proximity check). The length of the totp can change according to a server configuration.
+ We should probably add to the [documentation](https://github.com/wultra/powerauth-webflow/blob/develop/docs/Off-line-Signatures-QR-Code.md#forward-compatibility) that numerical value before nonce is treated as totp?